src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Cache.Memory;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="TagHelper"/> implementation targeting &lt;cache&gt; elements.
+    /// </summary>
+    public class CacheTagHelper : TagHelper
+    {
+        private const string VaryByAttributeName = "vary-by";
+        private const string VaryByHeaderAttributeName = "vary-by-header";
+        private const string VaryByQueryAttributeName = "vary-by-query";
+        private const string VaryByRouteAttributeName = "vary-by-route";
+        private const string VaryByCookieAttributeName = "vary-by-cookie";
+        private const string VaryByUserAttributeName = "vary-by-user";
+        private const string ExpiresOnAttributeName = "expires-on";
+        private const string ExpiresAfterAttributeName = "expires-after";
+        private const string ExpiresSlidingAttributeName = "expires-sliding";
+        private const string CachePriorityAttributeName = "priority";
+        private const string CacheKeyTokenSeparator = "||";
+        private static readonly char[] AttributeSeparator = new[] { ',' };
+
+        /// <summary>
+        /// Gets or sets the <see cref="IMemoryCache"/> instance used to cache entries.
+        /// </summary>
+        [Activate]
+        protected internal IMemoryCache MemoryCache { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ViewContext"/> for the current executing View.
+        /// </summary>
+        [Activate]
+        protected internal ViewContext ViewContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="string" /> to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByAttributeName)]
```

Comment:
Spoke to @NTaylorMullen and apparently the decision was to not make this change in this iteration.